### PR TITLE
demos: align C++ and Python IE configs

### PR DESCRIPTION
**Pull request by Wovchena**
Don't bind threads: thread binding came from benchmark_app which
does't run anything during inference. But if -d MULTI:CPU,GPU is
set, there is an extra thread to maintain GPU. In that case threads
aren't bound. Following that logic demos never bind threads because
there is main thread running pre and postprocessing. Additionally,
that's what IE says:
> 'YES' (default) binding option maps threads to cores - this
works best for static/synthetic scenarios like benchmarks.

Fix Python IE configs and follow C++ implementation.

### CODE REVIEW

File: demos/common/python/pipelines/async_pipeline.py [hunk 1]
```diff
@@ -17,6 +17,67 @@
...
+                result[device_value_list[0]] = int(device_value_list[1])
+        elif len(device_value_list) == 1 and device_value_list[0] != '':
+            for device in devices:
+                result[device] = int(device_value_list[0])
+        elif device_value_list[0] != '':
+            raise RuntimeError(f'Unknown string format: {values_string}')
+    return result
+
+
+def get_user_configs(flags_d: str, flags_nstreams: str, flags_nthreads: int)-> Dict[str, str]:
```
Comment (eizamaliev) [inline demos/common/python/pipelines/async_pipeline.py:56]: I'd prefer the old name, `get_plugin_configs`.

Comment (Wovchena) [reply demos/common/python/pipelines/async_pipeline.py:56]: The term plugin is hidden in IE implementation now. I don't thing it is a good idea to mention it.

File: demos/common/python/pipelines/async_pipeline.py [hunk 2]
```diff
@@ -27,6 +88,10 @@ def __init__(self, ie, model, plugin_config, device='CPU', max_num_requests=1):
         self.logger.info('Loading network to {} plugin...'.format(device))
         self.exec_net = ie.load_network(network=self.model.net, device_name=device,
                                         config=plugin_config, num_requests=max_num_requests)
+        if max_num_requests == 0:
+            # +1 to use it as a buffer of the pipeline
+            self.exec_net = ie.load_network(network=self.model.net, device_name=device,
+                                            config=plugin_config, num_requests=len(self.exec_net.requests) + 1)
```
Comment (eizamaliev) [inline demos/common/python/pipelines/async_pipeline.py:94]: Why do you load net twice? May  be `if ... else ...` more suitable here.

Comment (Wovchena) [reply demos/common/python/pipelines/async_pipeline.py:94]: There is no other way to query a suggestion about nireq in python

File: demos/common/python/pipelines/async_pipeline.py (hunk 1 above)
Comment (eizamaliev) [reply demos/common/python/pipelines/async_pipeline.py:56]: We could (and should) use brand new implementation in all py demos now. And not all demo have a pipeline approach and configure the IE manually.

Comment (Wovchena) [reply demos/common/python/pipelines/async_pipeline.py:56]: Straightforward pipelines wont' utilize more than one stream or req. `get_user_configs()` is supposed to be used with `AsyncPipeline` similar to C++.

File: demos/common/python/pipelines/async_pipeline.py (hunk 2 above)
Comment (eizamaliev) [reply demos/common/python/pipelines/async_pipeline.py:94]: You mean with `num_requests=0` the IE will determine it by d20? Ok, please add comment to this odd entity.

Comment (Wovchena) [reply demos/common/python/pipelines/async_pipeline.py]: What is d20? Added

**Comment by Wovchena** [comment]
Jenkins please retry a build

**Comment by Wovchena** [comment]
Jenkins please retry a build

**Comment by Wovchena** [comment]
Jenkins please retry a build

File: demos/common/python/pipelines/async_pipeline.py (hunk 2 above)
Comment (vladimir-dudnik) [reply demos/common/python/pipelines/async_pipeline.py]: is it lack of functionality in python api? is it make sense to submit bug on this?

Comment (Wovchena) [reply demos/common/python/pipelines/async_pipeline.py]: It is well described: https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx#L972 so it's not a bug, but lack of functionality.

**Review by vladimir-dudnik — APPROVED** [review]

File: demos/common/python/pipelines/async_pipeline.py (hunk 2 above)
Comment (vladimir-dudnik) [reply demos/common/python/pipelines/async_pipeline.py]: @ilya-lavrenov is it ok, to require customer to call load_network twice to query suggested nireq ?

Comment (ilya-lavrenov) [reply demos/common/python/pipelines/async_pipeline.py]: CC @akuporos @nkogteva 

I don't think that it's OK (at least C++ allows to create as many infer requests as you can), but looks like it's python limitations.

**Review by eizamaliev — APPROVED** [review]

**Pull request merged**